### ISSUE TITLE
Improved IIT Grantha font

### ISSUE THREAD

**Issue by virtualvinodh** [deepestblue/Sampradaya#28]
Hallo,

Once upon a time in a galaxy far far away, I worked on improving the IIT M Grantha font to make it legible and support conjuncts and Vedic svaras.

Few sample renderings can be found below.

[Grantha Sample - 1.pdf](https://github.com/deepestblue/Sampradaya/files/5766897/Grantha.Sample.-.1.pdf)
[Grantha Sample 2.pdf](https://github.com/deepestblue/Sampradaya/files/5766898/Grantha.Sample.2.pdf)
[Uttara Karana.pdf](https://github.com/deepestblue/Sampradaya/files/5766899/Uttara.Karana.pdf)

I added a bunch of glyphs and fixed some of the existing glyphs. I am attaching the font just in case it might be useful to you and you could reuse some of the glyphs.

[Grantha final modification 49.zip](https://github.com/deepestblue/Sampradaya/files/5766915/Grantha.final.modification.49.zip)

**Comment by deepestblue** [comment]
@virtualvinodh It's an honour to hear from you! My interest in Grantha would have gone nowhere without your work.

I've spent some time cleaning up some of the glyphs but I'd love to reuse as much of your work as possible. Would you be comfortable if I used your glyphs and released them under SIL OFL 1.1? Thanks in advance.

**Comment by virtualvinodh** [comment]
@deepestblue Sure. That's why I shared them. 

Let me know if you need any help with testing or OT programming. I'll be glad to chip in.

I was also thinking that if you already have an existing workflow (with QA), we could perhaps adapt it to port other legacy Grantha fonts to Unicode as well. We have at least 7 more (legacy) fonts for Grantha. 

V

**Comment by deepestblue** [comment]
@virtualvinodh Yes, would love to collaborate; I do have lots of test-data in the test/cases/ directory, covering all kinds of Grantha requirements, and I think they should make it easier to port over other fonts.

For now, could you help with the following? Could you attach the Unicode text for the sample text PDFs in the zip? I want to make sure I'm importing the new glyphs into the right codepoints, and having the text would help.

**Comment by virtualvinodh** [comment]
[sample-1.txt](https://github.com/deepestblue/Sampradaya/files/5845476/sample-1.txt)
[sample-2.txt](https://github.com/deepestblue/Sampradaya/files/5845477/sample-2.txt)
[karanagama-4-verses.txt](https://github.com/deepestblue/Sampradaya/files/5845479/karanagama-4-verses.txt)

The Karanagama is quite large, I have just transcribed the first four verses.

Hope this helps!

V

**Comment by deepestblue** [comment]
Attached Karanagama rendered into PDF on Windows 10 with the latest font from `master`. Looks fine to me.

[karanagama-4-verses.pdf](https://github.com/deepestblue/Sampradaya/files/5861260/karanagama-4-verses.pdf)

**Comment by deepestblue** [comment]
@virtualvinodh A few take-aways.

1. Thanks for the font and more importantly, the test cases :-) Most of the text rendered fine, except for Vedic accents. I did reuse the Grantha udatta accent glyph (the crescent moon) in the vedic_accents branch.
2. It looks like Vedic accent support in Sampradaya needs work in a couple of ways: positioning on the base glyph, and vertical kerning. I've created a comprehensive test-case based on your sample files, and filed 2 bugs #32 and #33 based on those. Feel free to checkout the vedic_accents branch if you want to give them a shot. I have thoughts on how to fix them so happy to brainstorm as well.
3. We have a pretty comprehensive test-suite of glyph combinations, and three different rasterisers: MacOS/CoreText, MacOS/Qt5 and Windows/DWrite. My QA workflow is running the three rasterisers and comparing the output PNGs against the checked-in master images. It's really easy to split the rasterisation code into its own Github project, and then we could use it to develop and improve any new font.
4. You mention 7 new fonts: where are they? Are they all Open Source or Open Source-able?

We can use this bug to keep chatting or switch to email.
